graylog2-server/src/main/java/org/graylog2/dashboards/widgets/DashboardWidgetCreator.java

```diff
@@ -104,28 +105,35 @@ public DashboardWidget fromPersisted(Searches searches, BasicDBObject fields) th
             throw new InvalidRangeParametersException("range type not set");
         }
 
-        TimeRange timeRange;
-        switch (rangeType) {
-            case "relative":
-                timeRange = new RelativeRange((Integer) timerangeConfig.get("range"));
-                break;
-            case "keyword":
-                timeRange = new KeywordRange((String) timerangeConfig.get("keyword"));
-                break;
-            case "absolute":
-                String from = new DateTime(timerangeConfig.get("from"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
-                String to = new DateTime(timerangeConfig.get("to"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
+        final String widgetId = (String) fields.get(DashboardWidget.FIELD_ID);
 
-                timeRange = new AbsoluteRange(from, to);
-                break;
-            default:
-                throw new InvalidRangeParametersException("range_type not recognized");
+        TimeRange timeRange;
+        try {
+            switch (rangeType) {
+                case "relative":
+                    timeRange = new RelativeRange((Integer) timerangeConfig.get("range"));
+                    break;
+                case "keyword":
+                    timeRange = new KeywordRange((String) timerangeConfig.get("keyword"));
+                    break;
+                case "absolute":
+                    String from = new DateTime(timerangeConfig.get("from"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
+                    String to = new DateTime(timerangeConfig.get("to"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
+
+                    timeRange = new AbsoluteRange(from, to);
+                    break;
+                default:
+                    throw new InvalidRangeParametersException("range_type not recognized");
+            }
+        } catch (InvalidRangeParametersException e) {
+            LOG.error("Invalid time range provided on widget " + widgetId, e);
+            timeRange = null;
```

Comment:
I think this will lead to problems in `getPersistedConfig()` of the concrete widget classes.